src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenRefExtensionMethods.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Open Technologies, Inc.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class CodeGenRefExtensionMethods : CompilingTestBase
```

Comment:
>CodeGenRefExtensionMethods [](start = 17, length = 26)

I see no benefit in putting this tests in a separate file and into a different test assembly. We can verify IL along the success cases in RefExtensionMethodsTests.cs